### PR TITLE
feat: update hertz and adapt content_encoding header

### PR DESCRIPTION
**Pull request by Duslia**
#### What type of PR is this?

feat
#### What this PR does / why we need it (English/Chinese):
en: update hertz and adapt content_encoding header
zh: 更新 Hertz 并且适配 Hertz 的 content-encoding pr

#### Which issue(s) this PR fixes:
https://github.com/cloudwego/hertz/issues/531

### CODE REVIEW

**Review by li-jin-gou — APPROVED** [review]

File: write.go [hunk 1]
```diff
@@ -362,6 +362,8 @@ func encodeHeaders(enc *hpack.Encoder, h *protocol.ResponseHeader, keys []string
 	// }
 
 	encKV(enc, consts.HeaderServerLower, string(h.Server()))
+	encKV(enc, consts2.HeaderEncodingLower, string(h.ContentEncoding()))
```
Comment (welkeyever) [inline write.go]: Do we need process other special headers？

Comment (Duslia) [reply write.go]: No, other special header has been handled in `writeResHeaders`

**Comment by Duslia** [comment]
https://github.com/cloudwego/hertz/issues/531

**Comment by kolinfluence** [comment]
@welkeyever everything ok?

**Comment by kolinfluence** [comment]
@Duslia possible to delete the "Server" header? i've tried deletion and somehow the "Server" header is always showing. can u try deleting "Server" header?

**Comment by kolinfluence** [comment]
@Duslia please expedite this. i was testing/debugging for real long time.
this repo last updated 7th Dec 2022

**Comment by Duslia** [comment]
> @Duslia please expedite this. i was testing/debugging for real long time. this repo last updated 7th Dec 2022

done. You can try this branch.

**Comment by kolinfluence** [comment]
@Duslia do i do 
go get github.com/hertz-contrib/http2@adapt_content_encoding ?

sorry i hv never tried doing so before. when can we really used merge version?

File: write.go [hunk 2]
```diff
@@ -361,7 +361,19 @@ func encodeHeaders(enc *hpack.Encoder, h *protocol.ResponseHeader, keys []string
 	// 	keys = sorter.Entries(h)
 	// }
 
-	encKV(enc, consts.HeaderServerLower, string(h.Server()))
+	// Add special header that not in `writeResHeaders`
+	// `writeResHeaders` handles the special headers it considers,
+	// such as content-lengh, etc., and the remaining headers are uniformly
+	// encoded using encodeHeaders. But the special header of hertz is
+	// different from the special header of writeResHeader. There are
+	// several additional headers, so special handler here.
```
Comment (welkeyever) [inline write.go:369]: "several additional headers" -> Are there other headers that need to be treated specially here(except `ContentEncoding `)?

**Comment by kolinfluence** [comment]
this is crucial for h2 to work properly.
sorry for complaining coz i was testing since Dec 7th with this problem...

File: write.go (hunk 2 above)
Comment (Duslia) [reply write.go:369]: `Server`, `cookie`. I have checked other fields.

**Comment by li-jin-gou** [comment]
LGTM

**Pull request merged**